core/templates/dev/head/exploration_editor/EditorServices.js

```diff
@@ -1952,3 +2012,302 @@ oppia.factory('explorationWarningsService', [
+                }
+                if (!angular.equals(
+                  newValue.outcome.feedback, oldValue.outcome.feedback)) {
+                  answerGroupHtml += (
+                    '<p>Feedback: ' + newValue.outcome.feedback + '</p>');
+                }
+                if (!angular.equals
+                    (newValue.rule_specs, oldValue.rule_specs)) {
+                  var rulesList = makeRulesListHumanReadable(newValue);
+                  if (rulesList.length) {
+                    answerGroupHtml += ('<div>Rules: <div>');
+                    for (var rule in rulesList) {
+                      answerGroupHtml += (rulesList[rule]).html();
+                    }
+                  }
+                }
+                stateWiseEditsMapping[stateName].push(
+                  angular.element('<div>Edited answer group: </div>')
+                    .append(answerGroupHtml));
+              } else {
+                stateWiseEditsMapping[stateName].push(
+                  angular.element('<div>Deleted answer group</div>'));
+              }
+              break;
+
+            case 'default_outcome':
+              var defaultOutcomeChanges = getRelativeChangeToGroups(lostChange);
+              var defaultOutcomeHtml = '';
+              if (defaultOutcomeChanges === 'added') {
+                defaultOutcomeHtml += (
+                  '<p>Destination: ' + newValue.dest + '</p>');
+                defaultOutcomeHtml += (
+                  '<p>Feedback: ' + newValue.feedback + '</p>');
+                stateWiseEditsMapping[stateName].push(
+                  angular.element('<div>Added default outcome: </div>')
+                    .append(defaultOutcomeHtml));
+              } else if (defaultOutcomeChanges === 'edited') {
+                if (newValue.dest !== oldValue.dest) {
+                  defaultOutcomeHtml += (
+                    '<p>Destination: ' + newValue.dest + '</p>');
+                }
+                if (!angular.equals(
+                  newValue.feedback, oldValue.feedback)) {
+                  defaultOutcomeHtml += (
+                    '<p>Feedback: ' + newValue.feedback + '</p>');
+                }
+                stateWiseEditsMapping[stateName].push(
+                  angular.element('<div>Edited default outcome: </div>')
+                    .append(defaultOutcomeHtml));
+              } else {
+                stateWiseEditsMapping[stateName].push(
+                  angular.element('<div>Deleted default outcome</div>'));
+              }
+              break;
+          };
+          break;
+      }
+    });
+
+    for (var stateName in stateWiseEditsMapping) {
+      var stateChangesEl = angular.element(
+        '<li>Edits to state: ' + stateName + '</li>');
+      for (var stateEdit in stateWiseEditsMapping[stateName]) {
+        stateChangesEl.append(stateWiseEditsMapping[stateName][stateEdit]);
+      }
+      outerHtml.append(stateChangesEl);
+    }
+    console.log(outerHtml);
```

Comment:
You can delete these (we try to keep the console clean so it's easier to see errors).